sources/packages/backend/apps/api/src/services/application-bulk-withdrawal/application-bulk-withdrawal-text.models.ts

```diff
@@ -0,0 +1,124 @@
+import { Contains, IsDate, IsNumberString, Length } from "class-validator";
+import { APPLICATION_NUMBER_LENGTH } from "@sims/sims-db";
+import { IsValidSIN } from "../../utilities/class-validation";
+import { getDateOnlyFromFormat } from "@sims/utilities";
+
+export const DATE_FORMAT = "YYYYMMDD";
+/**
+ * user-friendly header names used in the text to be populated by the user.
+ * The text model parser uses this as a base to parse the text string into an object model.
+ */
+export const DataTextHeaders = {
+  recordType: "Record Type",
+  sin: "SIN",
+  applicationNumber: "Application Number",
+  withdrawalDate: "Withdrawal Date",
+};
+
+export enum RecordType {
+  ApplicationBulkWithdrawalHeaderRecordType = "1",
+  ApplicationBulkWithdrawalDataRecordType = "2",
+  ApplicationBulkWithdrawalFooterRecordType = "3",
+}
+
+export class ApplicationWithdrawalTextModel {
+  /**
+   * Data record type.
+   */
+  @Contains(RecordType.ApplicationBulkWithdrawalDataRecordType, {
```

Comment:
just to double check, can we do `match` of `RecordType.ApplicationBulkWithdrawalDataRecordType`

Reply:
Yes we can, but as RecordType.ApplicationBulkWithdrawalDataRecordType is as enum constant used contains because in the future if they add additional Recordtype for the data we can expand it. The right one to be used should be @Equals.